save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/codeeditor/EditorToolbar.kt

```diff
@@ -0,0 +1,132 @@
+@file:Suppress("HEADER_MISSING_IN_NON_SINGLE_CLASS_FILE")
+
+package com.saveourtool.save.frontend.components.basic.codeeditor
+
+import com.saveourtool.save.frontend.components.basic.cardComponent
+import com.saveourtool.save.frontend.externals.fontawesome.*
+import com.saveourtool.save.frontend.externals.reactace.AceModes
+import com.saveourtool.save.frontend.externals.reactace.AceThemes
+import com.saveourtool.save.frontend.utils.buttonBuilder
+import com.saveourtool.save.frontend.utils.selectorBuilder
+import com.saveourtool.save.frontend.utils.withUnusedArg
+import csstype.ClassName
+import react.ChildrenBuilder
+import react.dom.html.ButtonType
+import react.dom.html.ReactHTML.button
+import react.dom.html.ReactHTML.div
+
+private val toolbarCard = cardComponent(isBordered = true)
+
+/**
+ * @param selectedMode
+ * @param selectedTheme
+ * @param selectedFileType
+ * @param hasUncommittedChanges
+ * @param setSelectedMode
+ * @param setSelectedTheme
+ * @param onUploadChanges
+ * @param onReloadChanges
+ * @param onResultReload
+ * @param onRunExecution
+ * @param onFileTypeChange
+ */
+@Suppress(
+    "TOO_MANY_PARAMETERS",
+    "LongParameterList",
+    "TOO_LONG_FUNCTION",
+    "LongMethod",
+)
+fun ChildrenBuilder.displayCodeEditorToolbar(
+    selectedMode: AceModes,
+    selectedTheme: AceThemes,
+    selectedFileType: FileType?,
+    hasUncommittedChanges: Map<FileType, Boolean>,
+    setSelectedMode: (String) -> Unit,
+    setSelectedTheme: (String) -> Unit,
+    onUploadChanges: () -> Unit,
+    onReloadChanges: () -> Unit,
+    onResultReload: () -> Unit,
+    onRunExecution: () -> Unit,
+    onFileTypeChange: (FileType) -> Unit,
+) {
+    toolbarCard {
+        div {
+            className = ClassName("input-group")
+            div {
+                className = ClassName("input-group-prepend")
+
+                button {
```

Comment:
Could we extract these `button`'s  into some methods, receiving `faIcon` and `title` as arguments?

Reply:
Opened #1320, tried to do it in this PR but I didn't like the implementation I had. Will try to come up with a better idea.